Sources/Glob/GlobSearch.swift

```diff
@@ -205,16 +205,29 @@ private func search(
                 symbolicLinkDestination = nil
             }
             if isDirectory {
-                group.addTask {
-                    try await search(
-                        directory: foundPath,
-                        symbolicLinkDestination: symbolicLinkDestination,
-                        matching: matching,
-                        includingPropertiesForKeys: keys,
-                        skipHiddenFiles: skipHiddenFiles,
-                        relativePath: relativePath + "/",
-                        continuation: continuation
-                    )
+                // This check is to prevent infinite loops when a symbolic link points to
+                // an ancestor directory of the current path.
+                let shouldSkipDirectory =
+                    if let symbolicLinkDestination,
+                    pathHasAncestor(maybeChild: directory, maybeAncestor: symbolicLinkDestination)
```

Comment:
What about the following naming? I think it reads better
```suggestion
                    is(symbolicLinkDestination, ancestorOf: directory)
```

Reply:
Yeah, I like it. I'll rename it!

Reply:
Yeah, that's better. I'll rename it!